server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -123,6 +123,7 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion TIME_BASED_K_ORDERED_DOC_ID_BACKPORT = def(8_520_00_0, Version.LUCENE_9_12_0);
     public static final IndexVersion DEPRECATE_SOURCE_MODE_MAPPER = def(8_521_00_0, Version.LUCENE_9_12_0);
     public static final IndexVersion USE_SYNTHETIC_SOURCE_FOR_RECOVERY_BACKPORT = def(8_522_00_0, Version.LUCENE_9_12_0);
+    public static final IndexVersion UPGRADE_TO_LUCENE_9_12_1 = def(8_523_00_0, Version.LUCENE_9_12_1);
```

Comment:
oh good point, what do we need this for if it's unused, can you remind me?

Reply:
It is "statically unused" - there are no static references. However, it is used reflectively to set the value of `LATEST_DEFINED` - the reflective code walks the declared fields.

Reply:
gotcha